src/IO/examples/snappy_read_buffer.cpp

```diff
@@ -0,0 +1,42 @@
+#include <memory>
+#include <string>
+
+#include <snappy.h>
+
+#include <IO/ReadBufferFromString.h>
+#include <IO/SnappyReadBuffer.h>
+#include <IO/WriteHelpers.h>
+#include <IO/copyData.h>
+#include "Common/Exception.h"
+
+
+int main()
+{
+    using namespace DB;
```

Comment:
Can you please also add an integration test where snappy will be passed as compression method?
For example, if you have a ready compressed file, you can put it into container inside an integration test 
like here https://github.com/ClickHouse/ClickHouse/blob/9e496910a0ddbe7d75b911d2a5a9b34c75b852d6/tests/integration/test_library_bridge/test.py#L37 and then put it into hdfs via `pyhdfs` command `copy_from_local` .
example tests with pyhdfs examples are here https://github.com/ClickHouse/ClickHouse/blob/9e496910a0ddbe7d75b911d2a5a9b34c75b852d6/tests/integration/test_merge_tree_hdfs/test.py#L167
But the test should be put it `test_storage_hdfs/test.py`
Or similar can ba done with s3.

Reply:
ok.

Reply:
@kssenii functional test already done, pls refer to `tests/queries/0_stateless/01059_storage_file_compression.sh`

Reply:
done.